### DISCUSSION TITLE
"Part of" incorrectly treats a PR as "Fixes issue XYZ"

### DISCUSSION THREAD

**Discussion by navaronbracke** [community/community#56551] [Pull Requests]
### Select Topic Area

Bug

### Body

When a PR mentions the following phrase `Part of #xyz` where `xyz` is a Github Issue,
it states that the PR will fix the aforementioned issue.

This is incorrect, as only `fixes` or `resolves` (Capitalized or not) should do that.
The word "Fixes" is never mentioned in the PR description.

Repro: https://github.com/SAP/gigya-flutter-plugin/pull/45

In this PR, it is part of issue 44, but it never states that it fixes it. However, issue 44 is listed as `will be fixed by` on the right-hand side.
When going to issue 44, it also states that it will be fixed by this PR at the top.

The phrase "Part of #xyz" is only intended to link to an issue (which still works as intended)

The funny part is, for the other PR's linked to the umbrella issue above, it works as expected. It that because those are drafts?

**Comment by hrishikesh-kocharekar-dev** [comment]
there is some confusion regarding the usage of the phrase "Part of #xyz" in a pull request (PR) description and its relation to fixing an issue on GitHub.

You are correct that the phrase "Part of #xyz" is typically used to link the PR to a specific GitHub issue, where "xyz" represents the issue number. It serves as a way to associate the PR with the issue and provide additional context. However, the phrase itself does not indicate that the PR will fix the mentioned issue.

In GitHub, there are specific keywords that can be used to associate a PR with an issue and indicate that the PR will fix the issue. These keywords are "Fixes," "Closes," or "Resolves" (case-insensitive). By using one of these keywords followed by the issue number, GitHub automatically links the PR to the issue and closes it once the PR is merged. For example, using "Fixes #xyz" in the PR description will both link the PR to issue "xyz" and close the issue once the PR is merged.

Regarding the situation you described with the PR and issue #44, it seems like the connection between the PR and the issue is established correctly. The issue correctly mentions that it will be fixed by the PR at the top, and the PR description includes the phrase "Part of #xyz" to link to the issue. Even though the PR description doesn't explicitly state that it fixes the issue, the connection is still established.

As for the other PRs linked to the same umbrella issue, if they are working as expected and properly closing the associated issues, it could be due to them using one of the recognized keywords ("Fixes," "Closes," or "Resolves") in their descriptions. The fact that they might be in draft status shouldn't affect the linkage between the PRs and the issues.

the phrase "Part of #xyz" in a PR description is primarily used to link the PR to a specific GitHub issue but doesn't indicate that the PR will fix the issue. To explicitly state that a PR fixes an issue, you should use one of the recognized keywords ("Fixes," "Closes," or "Resolves") followed by the issue number in the PR description.

**Reply by navaronbracke** [reply]
The problem is that the PR does *not* use the Fixes/Closes/Resolves phrases anywhere in the PR description, yet it still applies the "will be fixed by" relation to the parent issue. The intent of that PR is indeed to link to, but not fix entirely, the parent issue.

Is it because the parent issue uses the PR number in the table in its description? (where each item gets a PR linked, to make things easier to track)
